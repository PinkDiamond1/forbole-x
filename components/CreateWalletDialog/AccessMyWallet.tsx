import {
  Button,
  DialogActions,
  DialogContent,
  Typography,
  Box,
  TextField,
  DialogContentText,
  ButtonBase,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'

export enum ImportMode {
  ImportMnemonicPhrase = 'import mnemonic phrase',
  MnemonicPhraseBackup = 'use mnemonic phrase backup',
  ConnectLedgerDevice = 'connect ledger device',
}

interface AccessMyWalletProps {
  onConfirm(mode: ImportMode): void
}

const AccessMyWallet: React.FC<AccessMyWalletProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('access my wallet description')}</DialogContentText>
        <Box my={-2}>
          {Object.values(ImportMode).map((mode) => (
            <ButtonBase
              key={mode}
              className={classes.borderedButton}
              onClick={() => onConfirm(mode)}
            >
              <Typography variant="h6" gutterBottom>
                {t(mode)}
              </Typography>
              <Typography variant="body2" color="textSecondary">
                {t(`${mode} description`)}
              </Typography>
            </ButtonBase>
          ))}
        </Box>
      </DialogContent>
      <DialogActions>
        <Button className={classes.button} color="secondary">
          {t('create a wallet')}
        </Button>
      </DialogActions>
    </>
  )
}

export default AccessMyWallet
