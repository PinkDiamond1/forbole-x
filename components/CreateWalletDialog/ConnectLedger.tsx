import { Box, DialogContent, DialogContentText, Typography } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import LedgerImage from '../../assets/images/ledger.svg'
import useStyles from './styles'

interface ConnectLedgerProps {
  onConnect(address: string): void
}

const ConnectLedger: React.FC<ConnectLedgerProps> = () => {
  const { t } = useTranslation('common')
  const classes = useStyles()

  return (
    <DialogContent className={classes.dialogContent}>
      <DialogContentText>{t('connect ledger description')}</DialogContentText>
      <Box display="flex" flexDirection="column" alignItems="center" mb={6}>
        <LedgerImage />
        <Box mt={4}>
          <Typography>{t('connect ledger instruction')}</Typography>
        </Box>
      </Box>
    </DialogContent>
  )
}

export default ConnectLedger
