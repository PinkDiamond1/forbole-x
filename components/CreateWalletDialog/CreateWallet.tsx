import {
  ButtonBase,
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Grid,
  Typography,
  Box,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import MnemonicPhraseInput from '../MnemonicPhraseInput'
import useStyles from './styles'

interface CreateWalletProps {}

const CreateWallet: React.FC<CreateWalletProps> = () => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [mnemonic, setMnemonic] = React.useState('')

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('create new wallet description')}</DialogContentText>
        <MnemonicPhraseInput disabled mnemonic={mnemonic} onChange={setMnemonic} />
        <Box mt={6} />
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            onClick={() => null}
          >
            {t('create new wallet button')}
          </Button>
          <Typography align="center">{t('create new wallet caption')}</Typography>
        </Box>
      </DialogActions>
    </>
  )
}

export default CreateWallet
