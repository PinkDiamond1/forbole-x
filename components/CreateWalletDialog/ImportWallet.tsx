import { Button, DialogActions, DialogContent, Typography, Box, TextField } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'

interface ImportWalletProps {
  onConfirm(password: string): void
}

const ImportWallet: React.FC<ImportWalletProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [name, setName] = React.useState('')

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <Typography>{t('moniker')}</Typography>
        <TextField
          fullWidth
          variant="filled"
          placeholder={t('wallet name')}
          value={name}
          onChange={(e) => setName(e.target.value)}
          InputProps={{
            disableUnderline: true,
          }}
        />
        <Box mb={30} />
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={!name}
            onClick={() => onConfirm(name)}
          >
            {t('import')}
          </Button>
        </Box>
      </DialogActions>
    </>
  )
}

export default ImportWallet
