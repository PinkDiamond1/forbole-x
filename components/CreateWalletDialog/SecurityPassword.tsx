import {
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Typography,
  Box,
  useTheme,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import passwordStrength from 'check-password-strength'
import PasswordInput from '../PasswordInput'
import useStyles from './styles'

interface SecurityPasswordProps {
  onConfirm(password: string): void
}

const SecurityPassword: React.FC<SecurityPasswordProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const theme = useTheme()
  const [password, setPassword] = React.useState('')
  const passwordSecurityLevel = React.useMemo(() => {
    return password ? passwordStrength(password).id : 0
  }, [password])
  const passwordSecurityColors = [
    theme.palette.error.main,
    theme.palette.warning.main,
    theme.palette.success.main,
  ]

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('security password description')}</DialogContentText>
        <PasswordInput
          placeholder={t('password')}
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <Box mt={2} mb={30}>
          <Box my={2} display="flex" alignItems="center">
            <Box flex={1} height={theme.spacing(0.5)} bgcolor={passwordSecurityColors[0]} />
            <Box
              flex={1}
              height={theme.spacing(0.5)}
              bgcolor={
                passwordSecurityLevel > 0 ? passwordSecurityColors[1] : theme.palette.grey[50]
              }
            />
            <Box
              flex={1}
              height={theme.spacing(0.5)}
              bgcolor={
                passwordSecurityLevel > 1 ? passwordSecurityColors[2] : theme.palette.grey[50]
              }
            />
            <Typography
              style={{
                margin: theme.spacing(0, 1),
                color: passwordSecurityColors[passwordSecurityLevel],
              }}
              variant="body2"
            >
              {t(`password security level ${passwordSecurityLevel}`)}
            </Typography>
          </Box>
          <Typography variant="body2">{t('password caption')}</Typography>
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={password.length < 6}
            onClick={() => onConfirm(password)}
          >
            {t('next')}
          </Button>
        </Box>
      </DialogActions>
    </>
  )
}

export default SecurityPassword
