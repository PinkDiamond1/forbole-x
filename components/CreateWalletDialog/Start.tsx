import {
  ButtonBase,
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Grid,
  Typography,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'

interface StartProps {
  onImportWalletClick(): void
  onCreateWalletClick(): void
}

const Start: React.FC<StartProps> = ({ onImportWalletClick, onCreateWalletClick }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('create wallet description')}</DialogContentText>
        <Grid container spacing={5}>
          <Grid item md={6}>
            <ButtonBase className={classes.selectionBox} onClick={onImportWalletClick}>
              <Typography variant="body2" align="center" color="textSecondary">
                {t('i have mnemonic phrase')}
              </Typography>
            </ButtonBase>
          </Grid>
          <Grid item md={6}>
            <ButtonBase className={classes.selectionBox} onClick={onCreateWalletClick}>
              <Typography variant="body2" align="center" color="textSecondary">
                {t('i dont have mnemonic phrase')}
              </Typography>
            </ButtonBase>
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button className={classes.button} color="secondary">
          {t('what is mnemonic phrase')}
        </Button>
      </DialogActions>
    </>
  )
}

export default Start
