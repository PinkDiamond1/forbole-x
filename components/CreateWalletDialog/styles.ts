import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
  button: {
    flex: 1,
    margin: theme.spacing(2),
  },
  closeButton: {
    position: 'absolute',
    top: theme.spacing(2),
    right: theme.spacing(2),
  },
  backButton: {
    position: 'absolute',
    top: theme.spacing(2),
    left: theme.spacing(2),
  },
  selectionBox: {
    border: `1px solid ${theme.palette.grey[200]}`,
    borderRadius: theme.shape.borderRadius,
    height: theme.spacing(38),
    padding: theme.spacing(2, 4),
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
    '&:hover': {
      border: `1px solid ${theme.palette.grey[300]}`,
    },
  },
  dialogContent: {
    overflowY: 'initial',
  },
}))

export default useStyles
