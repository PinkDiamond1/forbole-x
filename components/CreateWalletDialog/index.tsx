import { Dialog, DialogTitle, IconButton } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import ClossIcon from '../../assets/images/icons/icon_cross.svg'
import React from 'react'
import useStyles from './styles'
import useIconProps from '../../misc/useIconProps'
import Start from './Start'
import CreateWallet from './CreateWallet'

type Stage = 'start' | 'import wallets' | 'import mnemonic' | 'create wallet' | 'confirm mnemonic'

interface CreateWalletDialogProps {
  open: boolean
  onClose(): void
}

const CreateWalletDialog: React.FC<CreateWalletDialogProps> = ({ open, onClose }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const iconProps = useIconProps()
  const [stage, setStage] = React.useState<Stage>('start')

  const content = React.useMemo(() => {
    switch (stage) {
      case 'start':
        return {
          title: t('create wallet title'),
          content: (
            <Start
              onCreateWalletClick={() => setStage('create wallet')}
              onImportWalletClick={() => setStage('import wallets')}
            />
          ),
        }
      case 'import wallets':
        return {
          title: '',
          content: <></>,
        }
      case 'import mnemonic':
        return {
          title: '',
          content: <></>,
        }
      case 'create wallet':
        return {
          title: t('create new wallet title'),
          content: <CreateWallet />,
        }
      case 'confirm mnemonic':
        return {
          title: '',
          content: <></>,
        }
      default:
        return {
          title: '',
          content: <></>,
        }
    }
  }, [stage, t])

  return (
    <Dialog fullWidth open={open} onClose={onClose}>
      <IconButton className={classes.closeButton} onClick={onClose}>
        <ClossIcon {...iconProps} />
      </IconButton>
      <DialogTitle>{content.title}</DialogTitle>
      {content.content}
    </Dialog>
  )
}

export default CreateWalletDialog
