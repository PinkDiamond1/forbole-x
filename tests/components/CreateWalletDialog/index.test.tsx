import React from 'react'
import renderer from 'react-test-renderer'
import { Secp256k1HdWallet } from '@cosmjs/launchpad'
import CreateWalletDialog from '../../../components/CreateWalletDialog'

const mockWalletsContext = {
  addWallet: jest.fn(),
}

const onClose = jest.fn()

jest.mock('../../../contexts/WalletsContext', () => ({
  useWalletsContext: () => mockWalletsContext,
}))

jest.mock('../../../components/CreateWalletDialog/Start', () => (props) => (
  <div id="Start" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/CreateWallet', () => (props) => (
  <div id="CreateWallet" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/ConfirmMnemonic', () => (props) => (
  <div id="ConfirmMnemonic" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/SecurityPassword', () => (props) => (
  <div id="SecurityPassword" {...props} />
))
jest.mock('../../../components/CreateWalletDialog/ImportWallet', () => (props) => (
  <div id="ImportWallet" {...props} />
))
jest.mock('@material-ui/core/Dialog', () => (props) => <div id="dialog" {...props} />)
jest.mock('@cosmjs/launchpad', () => ({
  Secp256k1HdWallet: {
    generate: jest.fn().mockResolvedValue({
      mnemonic: 'guide check kick present flash casual',
      pubkey: { 1: { 2: 3 } },
    }),
  },
}))

describe('component: CreateWalletDialog', () => {
  it('renders open state correctly', () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders open state correctly', () => {
    const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders import wallets stage correctly', () => {
    const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
    renderer.act(() => {
      component.root.findByProps({ id: 'Start' }).props.onImportWalletClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when back button is clicked on import wallets stage', () => {
    const component = renderer.create(<CreateWalletDialog open={false} onClose={onClose} />)
    renderer.act(() => {
      component.root.findByProps({ id: 'Start' }).props.onImportWalletClick()
    })
    renderer.act(() => {
      component.root
        .findByProps({ 'data-file-name': 'SvgIcon_back' })
        .parent.parent.parent.props.onClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onClose when close icon button is clicked', () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    renderer.act(() => {
      component.root
        .findByProps({ 'data-file-name': 'SvgIcon_cross' })
        .parent.parent.parent.props.onClick()
    })
    expect(onClose).toBeCalled()
  })
  it('calls Secp256k1HdWallet.generate(24) and renders create wallet stage correctly when onCreateWalletClick is called', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    expect(Secp256k1HdWallet.generate).toBeCalledWith(24)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders confirm mnemonic stage correctly when onConfirm is called from CreateWallet', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'CreateWallet' }).props.onConfirm()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders set security password stage correctly when mnemonic is confirmed to be correct', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    const { mnemonic: mockMnemonic } = await Secp256k1HdWallet.generate()
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'CreateWallet' }).props.onConfirm()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'ConfirmMnemonic' }).props.onConfirm(mockMnemonic)
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when confirm mnemonic is invalid', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'CreateWallet' }).props.onConfirm()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'ConfirmMnemonic' }).props.onConfirm('invalid mnemonic')
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders import wallet stage correctly when security password is set', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    const { mnemonic: mockMnemonic } = await Secp256k1HdWallet.generate()
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'CreateWallet' }).props.onConfirm()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'ConfirmMnemonic' }).props.onConfirm(mockMnemonic)
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'SecurityPassword' }).props.onConfirm('123123')
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls addWallet and onClose when wallet name is set', async () => {
    const component = renderer.create(<CreateWalletDialog open onClose={onClose} />)
    const { mnemonic: mockMnemonic, pubkey: mockPubkey }: any = await Secp256k1HdWallet.generate()
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'Start' }).props.onCreateWalletClick()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'CreateWallet' }).props.onConfirm()
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'ConfirmMnemonic' }).props.onConfirm(mockMnemonic)
    })
    renderer.act(() => {
      component.root.findByProps({ id: 'SecurityPassword' }).props.onConfirm('123123')
    })
    await renderer.act(async () => {
      await component.root.findByProps({ id: 'ImportWallet' }).props.onConfirm('wallet name')
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(mockWalletsContext.addWallet).toBeCalledWith({
      mnemonic: mockMnemonic,
      name: 'wallet name',
      securityPassword: '123123',
      pubkey: mockPubkey,
    })
    expect(onClose).toBeCalled()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
