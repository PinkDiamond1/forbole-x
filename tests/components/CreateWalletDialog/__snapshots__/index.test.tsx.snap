// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog renders closed state correctly 1`] = `
<div
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  onClose={[Function]}
  open={false}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:create wallet title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-9"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create wallet description
    </p>
    <div
      className="MuiBox-root MuiBox-root-18"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-19"
            >
              <svg
                data-file-name="SvgLogin_light"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              common:import wallet
            </p>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-20"
            >
              <svg
                data-file-name="SvgCreate_wallet_light"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              common:create wallet
            </p>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  />
  <div
    className="MuiBox-root MuiBox-root-21"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(0%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-22"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-23"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-24"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-25"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`component: CreateWalletDialog renders open state correctly 1`] = `
<div
  fullScreen={false}
  fullWidth={true}
  id="dialog"
  onClose={[Function]}
  open={true}
>
  <button
    className="MuiButtonBase-root MuiIconButton-root HookGlobalStyles-closeButton-2"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_cross"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
  </button>
  <div
    className="MuiDialogTitle-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h6"
    >
      common:create wallet title
    </h2>
  </div>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-9"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:create wallet description
    </p>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-11"
            >
              <svg
                data-file-name="SvgLogin_light"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              common:import wallet
            </p>
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-selectionBox-4"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-12"
            >
              <svg
                data-file-name="SvgCreate_wallet_light"
              />
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
            >
              common:create wallet
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  />
  <div
    className="MuiBox-root MuiBox-root-13"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={100}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(0%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-14"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-15"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-16"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <div
      className="MuiBox-root MuiBox-root-17"
    />
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
      role="progressbar"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
  </div>
</div>
`;
